src/test/kotlin/com/pubnub/contract/state/TestSinkSource.kt

```diff
@@ -0,0 +1,35 @@
+package com.pubnub.contract.state
+
+import com.fasterxml.jackson.databind.PropertyNamingStrategies.SnakeCaseStrategy
+import com.pubnub.api.eventengine.EffectInvocation
+import com.pubnub.api.eventengine.Event
+import com.pubnub.api.eventengine.QueueSinkSource
+import com.pubnub.api.eventengine.Sink
+import com.pubnub.api.eventengine.Source
+import java.util.concurrent.LinkedBlockingQueue
+
+class TestSinkSource<T>(
+    private val testSink: MutableList<Pair<String, String>>,
+    private val sinkSource: QueueSinkSource<T> = QueueSinkSource(LinkedBlockingQueue())
+) : Sink<T>, Source<T> by sinkSource {
+
+    private val snakeCaseStrategy: SnakeCaseStrategy = SnakeCaseStrategy()
+    override fun add(t: T) {
+        testSink.add(t.type() to t.name())
+        sinkSource.add(t)
+    }
+
+    override fun take(): T {
+        return sinkSource.take()
+    }
```

Comment:
This can be omitted. It's covered by `by sinkSource`

Reply:
ok